### PR TITLE
Record volumeID in GlusterFS PV spec.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@jsafrane Please review :). Thanks

**Comment by jsafrane** [comment]
What's the difference between `volume.Name` (a.k.a. `GlusterfsVolumeSource.Path`) and `volume.Id`? How is it going to be used?

**Comment by humblec** [comment]
@jsafrane first of all, sorry for not making it clear in the initial comment. This patch is a background work for supporting custom volume names in the provisioner. Heketi always work on `volume IDs` , for ex: if we want to delete a volume, the argument can be only `volume ID` . Its strightforward to retrieve `volume ID` as long as the name is chosen by heketi. It appends `vol_` as a prefix to `volume ID` and form the volume name. For ex: `vol_123-123..`. If the volume name is custom, its not possible to retrieve ID by this way. The plan is to also record the volume ID in the spec, so that at time of deletion we can use that ID .  Old existing volume will also work as I am planning to do the existing mechanism if the spec dont have the `volID`. Hope this clear the dust.

**Comment by humblec** [comment]
@jsafrane PTAL :)

File: pkg/volume/glusterfs/glusterfs.go [hunk 1]
```diff
@@ -721,6 +721,7 @@ func (p *glusterfsVolumeProvisioner) Provision() (*v1.PersistentVolume, error) {
 		glusterTypeAnn:                             "file",
 		"Description":                              glusterDescAnn,
 		v1.MountOptionAnnotation:                   "auto_unmount",
+		"VolID":                                    volID,
```
Comment (jsafrane) [inline pkg/volume/glusterfs/glusterfs.go]: This should be some `gluster.kubernetes.io/volume-id` or something like that.

**Review by jsafrane — COMMENTED** [review]
If the ID is going to be more important that volume name, should it be part of `GlusterfsVolumeSource`? When it's set, it's used to mount & delete & resize the volume, if it's not set then `GlusterfsVolumeSource.Path` must have `vol_<ID>` syntax?

File: pkg/volume/glusterfs/glusterfs.go (hunk 1 above)
Comment (humblec) [reply pkg/volume/glusterfs/glusterfs.go]: Changed it to `gluster.kubernetes.io/heketi-volume-id` which is more conveying.

**Comment by humblec** [comment]
@jsafrane the volID is important for heketi and not for Gluster, so placing it in the glusterfs volume source is not required and gluster mount requires volume name. As mentioned earlier, based on `volid` has set in pv spec, I will be operating on heketi calls like `delete` and `resize`.  Addressed your comment. PTAL. Thanks !

**Comment by humblec** [comment]
@jsafrane @jingxu97  any idea why this is not yet merged ? its more than 10 days it wait here with lgtm and approval.

**Comment by jsafrane** [comment]
We're frozen for 1.9. Be patient. And ready for rebases once the flood gates open.
Link to "Submit Queue" in this PR will always tell you what's wrong, this time it's "Milestone is for a future release and cannot be merged".

**Comment by humblec** [comment]
@jsafrane Sure, yeah, will wait for the door to open :). Thanks

**Comment by humblec** [comment]
@jsafrane had to rebase as expected :), can you bring back lgtm on this PR ?

**Comment by humblec** [comment]
@jsafrane PTAL :). Thanks

**Comment by humblec** [comment]
@jsafrane @jingxu97 Can you please bring back LGTM here ?

**Pull request merged**